node/src/components/block_synchronizer/deploy_acquisition/tests.rs

```diff
@@ -0,0 +1,176 @@
+use std::collections::{BTreeMap, VecDeque};
+
+use crate::types::{Block, Deploy};
+use assert_matches::assert_matches;
+use casper_execution_engine::storage::trie::merkle_proof::TrieMerkleProof;
+use casper_types::{testing::TestRng, AccessRights, CLValue, StoredValue, URef};
+use rand::Rng;
+
+use super::*;
+
+fn gen_test_deploys(rng: &mut TestRng) -> BTreeMap<DeployHash, Deploy> {
+    let num_deploys = rng.gen_range(2..15);
+    (0..num_deploys)
+        .into_iter()
+        .map(|_| {
+            let deploy = Deploy::random(rng);
+            (*deploy.hash(), deploy)
+        })
+        .collect()
+}
+
+fn gen_approvals_hashes<'a, I: Iterator<Item = &'a Deploy> + Clone>(
+    rng: &mut TestRng,
+    deploys_iter: I,
+) -> ApprovalsHashes {
+    let block = Block::random_with_deploys(rng, deploys_iter.clone());
+    ApprovalsHashes::new(
+        block.hash(),
+        deploys_iter
+            .map(|deploy| deploy.approvals_hash().unwrap())
+            .collect(),
+        TrieMerkleProof::new(
+            URef::new([255; 32], AccessRights::NONE).into(),
+            StoredValue::CLValue(CLValue::from_t(()).unwrap()),
+            VecDeque::new(),
+        ),
+    )
+}
+
+#[test]
+fn dont_apply_approvals_hashes_when_acquiring_by_id() {
+    let mut rng = TestRng::new();
+    let test_deploys = gen_test_deploys(&mut rng);
+    let approvals_hashes = gen_approvals_hashes(&mut rng, test_deploys.values());
+
+    let mut deploy_acquisition = DeployAcquisition::ById(Acquisition::new(
+        test_deploys
+            .iter()
+            .map(|(deploy_hash, deploy)| {
+                DeployId::new(*deploy_hash, deploy.approvals_hash().unwrap())
+            })
+            .collect(),
+        false,
+    ));
+
+    assert_matches!(
+        deploy_acquisition.apply_approvals_hashes(&approvals_hashes),
+        Err(Error::AcquisitionByIdNotPossible)
```

Comment:
Can you add a test covering the [EncounteredNonVacantDeployState](https://github.com/casper-network/casper-node/blob/dev/node/src/components/block_synchronizer/deploy_acquisition.rs#L15) error variant?

Reply:
added here: https://github.com/casper-network/casper-node/pull/3690/commits/c5117c8e0139028ad0e64f020d7e8e06b04143cc